### DISCUSSION TITLE
livebuild not installing veracrypt .deb package

### DISCUSSION THREAD

**Discussion by manujchandra** [SpiralLinux/SpiralLinux-project#423] [Q&A] [answered]
Hi,

I am building a custom ISO using SpiralLinux 12. I added 2 Debian packages in `SpiralLinux-XFCE/config/packages` folder.

One is [VSCodium](https://github.com/VSCodium/vscodium/releases/download/1.88.1.24104/codium_1.88.1.24104_amd64.deb) and the other is [Veracrypt](https://launchpad.net/veracrypt/trunk/1.26.7/+download/veracrypt-1.26.7-Debian-12-amd64.deb)

But after building the ISO, only VSCodium is installed and Veracrypt is ignored. I have even tried to rename the .deb package to veracrypt.deb but it still won't install.

[build.log](https://github.com/SpiralLinux/SpiralLinux-project/files/15091694/build.log)

In the logs above, I can see references to codium, but no mention of veracrypt. I have checked the permissions and both Debian packages have the same permission. The Veracrypt .deb file is being ignored. Any idea what might be the issue?

Thanks!

**Comment by geckolinux** [comment]
Hi there, maybe try installing Veracrypt manually in a fresh SpiralLinux system (you can just boot the live system) and see what happens when you try to install it manually, does it pull in any other dependencies and if so where do those come from?

**Reply by manujchandra** [reply]
See my other [post](https://github.com/orgs/SpiralLinux/discussions/424). I cannot access internet from guest, so I am unable to try to install it manually.

**Comment by manujchandra** [comment]
I ran the command `dpkg -I veracrypt-1.26.7-Debian-12-amd64.deb` and got the following output:

```
 new Debian package, version 2.0.
 size 9211094 bytes: control archive=12881 bytes.
     415 bytes,    12 lines      control              
   41414 bytes,   452 lines      md5sums              
     162 bytes,     9 lines      prerm                #!/bin/sh
 Architecture: amd64
 Conflicts: veracrypt-console
 Depends: libwxgtk3.2-1, libayatana-appindicator3-1, libfuse2, dmsetup, sudo, libpcsclite1, pcscd
 Description: Disk encryption with strong security based on TrueCrypt.
  Disk encryption with strong security based on TrueCrypt.
 Maintainer: VeraCrypt Team <veracrypt@idrix.fr>
 Package: veracrypt
 Priority: optional
 Section: libs
 Version: 1.26.7-1
 Installed-Size: 25830
```

I added the packages `libwxgtk3.2-1, libayatana-appindicator3-1, libfuse2, dmsetup, libpcsclite1, pcscd` to `spirallinux-XFCE-packages.list.chroot` but it still won't install.

I am investigating further.

**Comment by manujchandra** [comment, marked as answer]
I renamed `veracrypt-1.26.7-Debian-12-amd64.deb` to `veracrypt_1.26.7_Debian_12_amd64.deb` and for some reason it worked!

(replaced - with _)

**Reply by geckolinux** [reply]
Interesting, thanks for confirming the workaround.

**Reply by manujchandra** [reply]
It is my unconfirmed hypothesis that one hyphen is allowed in the name, but if there are more than one hyphen, it won't install.

```
codium_1.88.1.24104_amd64.deb
ueberzugpp_2.9.4_amd64.deb
veracrypt_1.26.7_Debian_12_amd64.deb
virtualbox-7.0_7.0.16-162802~Debian~bookworm_amd64.deb
```

I noticed that all other debs have 1 or less than 1 hyphens, and they were working. For example, VirtualBox has 1 hyphen and it works. My hypothesis is that if one more hyphen is added to the name, it won't install. Maybe that's why they used ~ instead of hyphens. But that's something to be tested on a rainy day.
